src/y0/algorithm/conditional_independencies.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+
+"""An implementation to get conditional independencies of an ADMG."""
+
+import copy
+from itertools import chain, combinations, groupby
+from typing import Iterable, Optional, Set, Tuple
+
+import networkx as nx
+from ananke.graphs import SG
+from tqdm import tqdm
+
+from ..struct import DSeparationJudgement
+from ..utils import powerset
+
+__all__ = [
+    'are_d_separated',
+    'get_conditional_independencies',
+]
+
+
+def get_conditional_independencies(
+    graph: SG,
```

Comment:
the reason we're using this style of definitions throughout the package and not the hanging one is that it's consistent in where the variables line up regardless of the length of the function name. the 4 space indent also is consistent with all of the other indentation in python code, while hanging indents make it much more difficult to read - that's why I've gone back and changed these again